### ISSUE TITLE
Lint/UnreachableLoop false positive when using conditional next

### ISSUE THREAD

**Issue by Tao-Galasse** [rubocop/rubocop#9746]
In a loop where I conditionally return a value or calling `next`, rubocop triggers the Lint/UnreachableLoop.

Here is an example : 
```ruby
%w[foo bar tao].each do |name|
  return User.find_by(first_name: name) || next
end
```
If I have a user named `tao` but no user named `foo` or `bar`, this will run 3 times.

--------

## Expected behavior

Not raising a `Lint/UnreachableLoop` warning.

## Actual behavior

```
W: Lint/UnreachableLoop: This loop will have at most one iteration.
      %w[foo bar tao].each do |name| ...
```

## Steps to reproduce the problem

You can use the example above. You can even replace `next` by something like `(puts 'next'; next)` to see it iterate.

## RuboCop version

1.13.0

--------

Thanks a lot for your work, you rock! ❤️
